core/src/main/scala/kafka/controller/PartitionStateMachine.scala

```diff
@@ -235,6 +268,58 @@ class PartitionStateMachine(controller: KafkaController) extends Logging {
     }
   }
 
+
+  /**
+    * Asynchronous version of handleStateChange only for the case when targetState == OnlinePartition.
+    * For any other transitions the syncronous version should be used
+    */
+  private def handleStateChangeAsync(topic: String, partition: Int, targetState: PartitionState,
```

Comment:
The code is pretty much all duplicated from the sync version. Can't we either make it all asynchronous or at least do a check only for the new lines to avoid duplication?